app/dashboard/management/commands/sync_es_profile.py

```diff
@@ -0,0 +1,63 @@
+'''
+    Copyright (C) 2019 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+import time
+
+from django.core.management.base import BaseCommand
+
+# https://github.com/gitcoinco/web/issues/4424
+
+class Command(BaseCommand):
+
+    help = 'syncs email subscriber and profiles'
+
+    def handle(self, *args, **options):
+        from dashboard.models import Profile, SearchHistory
+        from marketing.models import EmailSubscriber
+
+        counter = 0 
+        # pull keywords by search history
+        for profile in Profile.objects.filter(keywords=[]):
+
+            histories = SearchHistory.objects.filter(user__profile=profile)
+            keywords = []
+            for history in histories:
+                if history.data.get('keywords', None):
+                    keywords = history.data['keywords'][0].split(',')
+            if keywords:
+                print("1", counter, profile.handle, keywords)
+                counter += 1
+                profile.keywords = keywords
+                profile.save()
+
+        # pull keywords by emailsubscriber into profile
+        for profile in Profile.objects.filter(keywords=[]):
```

Comment:
Here we are iterating over the profiles that don't have any keywords, but the for loop above assigns keywords to each profile that lacks them if there is a search history object for that user. In this loop, we are combining the email subscriber keywords with the profile keywords, but we are only iterating over profiles that have no keywords.

I think we need to combine the two `for profile in Profile.objects.filter(keywords=[]):` loops so we can get both search history and email subscriber keywords applied. In the current code, adding the search history keywords to a profile essentially skips adding email subscriber keywords to that same profile.

Reply:
32a1a057affae79ff37beebcf5a6914c9d82f085